release-notes/8.0/8.0.0/api-diff/Microsoft.NETCore.App/8.0.0_System.Runtime.CompilerServices.md

```diff
@@ -0,0 +1,123 @@
++    public sealed class NullablePublicOnlyAttribute : Attribute {
++        public readonly bool IncludesInternals;
++        public NullablePublicOnlyAttribute(bool value);
++    }
++    [AttributeUsageAttribute(AttributeTargets.Module, Inherited=false)]
++    [EditorBrowsableAttribute(EditorBrowsableState.Never)]
++    public sealed class RefSafetyRulesAttribute : Attribute {
++        public RefSafetyRulesAttribute(int version);
++        public int Version { get; }
++    }
+-    [AttributeUsageAttribute(AttributeTargets.Class | AttributeTargets.Field | AttributeTargets.Property | AttributeTargets.Struct, AllowMultiple=false, Inherited=false)]
+-    public sealed class RequiredMemberAttribute : Attribute
++    [AttributeUsageAttribute(AttributeTargets.Class | AttributeTargets.Field | AttributeTargets.Property | AttributeTargets.Struct, AllowMultiple=false, Inherited=false)]
++    [EditorBrowsableAttribute(EditorBrowsableState.Never)]
++    public sealed class RequiredMemberAttribute : Attribute
++    [AttributeUsageAttribute(AttributeTargets.Parameter, Inherited=false)]
++    [EditorBrowsableAttribute(EditorBrowsableState.Never)]
++    public sealed class RequiresLocationAttribute : Attribute {
++        public RequiresLocationAttribute();
++    }
+     public sealed class RuntimeWrappedException : Exception {
+-        public override void GetObjectData(SerializationInfo info, StreamingContext context);
++        [EditorBrowsableAttribute(EditorBrowsableState.Never)]
++        [ObsoleteAttribute("This API supports obsolete formatter-based serialization. It should not be called or extended by application code.", DiagnosticId="SYSLIB0051", UrlFormat="https://aka.ms/dotnet-warnings/{0}")]
++        public override void GetObjectData(SerializationInfo info, StreamingContext context);
+     }
++    [AttributeUsageAttribute(AttributeTargets.Parameter, Inherited=false)]
++    [EditorBrowsableAttribute(EditorBrowsableState.Never)]
++    public sealed class ScopedRefAttribute : Attribute {
++        public ScopedRefAttribute();
++    }
+     public sealed class SwitchExpressionException : InvalidOperationException {
+-        public override void GetObjectData(SerializationInfo info, StreamingContext context);
++        [EditorBrowsableAttribute(EditorBrowsableState.Never)]
++        [ObsoleteAttribute("This API supports obsolete formatter-based serialization. It should not be called or extended by application code.", DiagnosticId="SYSLIB0051", UrlFormat="https://aka.ms/dotnet-warnings/{0}")]
++        public override void GetObjectData(SerializationInfo info, StreamingContext context);
+     }
+     public static class Unsafe {
+-        public static bool AreSame<T>([AllowNullAttribute] ref T left, [AllowNullAttribute] ref T right);
++        public static bool AreSame<T>([AllowNullAttribute, RequiresLocationAttribute] ref T left, [AllowNullAttribute, RequiresLocationAttribute] ref T right);
```

Comment:
@carlossanlop, `[RequiresLocation] ref T` should be emitted as `ref readonly` in the diff

Reply:
Opened an issue to track this request: https://github.com/dotnet/arcade/issues/14238